app/assets/javascripts/i18n.js

```diff
@@ -857,12 +854,43 @@
     }
 
     return scope;
-  }
+  };
+  /**
+   * Merge obj1 with obj2
+   * @param {Object} obj1 Default settings
+   * @param {Object} obj2 User obj2
+   * @returns {Object} Merged values of obj1 and obj2
+   */
+  I18n.extend = function ( obj1, obj2 ) {
+    var extended = {};
+    var prop;
+    for (prop in obj1) {
+      if (Object.prototype.hasOwnProperty.call(obj1, prop)) {
```

Comment:
Maybe there is some misunderstanding here.
I have read [the page](http://stackoverflow.com/questions/8157700/object-has-no-hasownproperty-method-i-e-its-undefined-ie8) and I agree there should be not issue with this function supporting ES3.

In additional to the code, if there is external sources of evidence supporting the decision of having this version of implementation, let's add them in comment.
Other dev can save some time by just visiting the URL, instead of having to find the line comment in this PR.

Doc + Test > Implementation